### PR TITLE
Doc versioning

### PR DESCRIPTION
**Pull request by gonzalocasas**
Use the newly added doc versioning option/widget of `sphinx-compas-theme 0.9` to allow switching to different versions of the documentation. I manually added some of the older doc builds to the doc github pages (only to be able to have something to switch on 😛 )

It now looks like this:
image

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [x] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.Configuration`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -68,6 +68,15 @@
     "package_title": project,
     "package_version": release,
     "package_repo": 'https://github.com/compas-dev/compas_fab',
+    "package_docs": 'https://gramaziokohler.github.io/compas_fab',
+    "package_old_versions": [
```
Comment (beverlylytle) [inline docs/conf.py:72]: Is it possible to put in a hook to update this on each new release?

**Comment by gonzalocasas** [comment]
I fixed the docs github actions workflow to publish versioned docs. Docs from pull requests are published to `https://gramaziokohler.github.io/compas_fab/pull_<###>`, tags are published to the tag name (eg `https://gramaziokohler.github.io/compas_fab/<tag_name>`) and main branch is published to `https://gramaziokohler.github.io/compas_fab/`latest`.

This PR in particular has been published to https://gramaziokohler.github.io/compas_fab/pull_196

File: docs/doc_versions.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+latest
+stable
```
Comment (beverlylytle) [inline docs/doc_versions.txt:2]: What does "stable" mean in the context of not yet having a Version 1.0?

File: docs/conf.py [hunk 3]
```diff
@@ -68,6 +76,8 @@
     "package_title": project,
     "package_version": release,
     "package_repo": 'https://github.com/compas-dev/compas_fab',
+    "package_docs": package_docs_root,
+    "package_old_versions": package_docs_versions
```
Comment (beverlylytle) [inline docs/conf.py:80]: Doesn't "package_old_versions" also contain "latest"?  Is that already old?

File: docs/doc_versions.txt (hunk 2 above)
Comment (gonzalocasas) [reply docs/doc_versions.txt:2]: `stable` is supposed to be the latest *released* version (vs `latest` being bleeding edge from main branch). The idea is to have a permalink into the newest release, but right now, no commit automatically generates it. 

I am not sure if we can use the `tag` event to push at the same time to `stable` and assume that "last tag == stable release"

Comment (beverlylytle) [reply docs/doc_versions.txt:2]: Ah, so the intention is that 0.11.0 should be the same thing as "stable"?

File: docs/conf.py (hunk 3 above)
Comment (gonzalocasas) [reply docs/conf.py:80]: Maybe a better name would have been `package_all_versions` 😛  (this variable name is [defined in the theme](https://github.com/compas-dev/sphinx_compas_theme/blob/master/sphinx_compas_theme/compaspkg/theme.conf))

File: docs/doc_versions.txt (hunk 2 above)
Comment (gonzalocasas) [reply docs/doc_versions.txt:2]: yep! I believe that's the way readthedocs does/names them as well.

**Review by beverlylytle — APPROVED** [review]

**Pull request merged**